### ISSUE TITLE
Bug in regular expression `[^\R]` does not match `R`.

### ISSUE THREAD

**Issue by hazi** [coteditor/CotEditor#1317]
### Description

There seems to be a bug in the search where `[^\R]` does not match only `R`.
This also affects the syntax color of JSON, etc. Only keys that contain R have a different color.

<img width="481" alt="image" src="https://user-images.githubusercontent.com/5656927/171299731-e1e03127-283c-4633-86da-f453be55584c.png">

### To Reproduce

Perform a search using a regular expression

### CotEditor version

4.2.2

### macOS version

12.4

**Comment by hazi** [comment]
I have confirmed the same behavior in Ruby.
It may be a library specification or a bug.

**Comment by hazi** [comment]
Was the library used an 鬼車?
It was confirmed that it was an invalid regular expression to begin with.
https://github.com/kkos/oniguruma/blob/d484916f3ca2b6bbc81accba63054625dfe26b6b/doc/RE#L83-L84

**Comment by 1024jp** [comment]
Thank you for the report.

CotEditor uses NSRegularExpression that uses the [ICU regular expression engine](https://unicode-org.github.io/icu/userguide/strings/regexp.html) for all regular expression procedures. So, it's currently not Oniguruma, though Oniguruma was actually used until CotEditor 3.0.0.

The issue you reported was simply my mistake. I did not check well if `\R` can be used in `[]` when I updated the styles for line ending handling update on CotEditor 4.2.0.
